### PR TITLE
Add a test for cookies with semicolon

### PR DESCRIPTION
**Pull request by dawehner**
In integration between Drupal and the phantomjs driver in mink (https://github.com/jcalderonzumba/MinkPhantomJSDriver)
we ran into an issue with cookies containing semicolons. This bug is probably due to missing url encoding in phantomjs itself
or some JS.

This PR is about ensuring that using semicolons actually work.

### CODE REVIEW

File: driver-testsuite/tests/Basic/CookieTest.php [hunk 1]
```diff
@@ -45,6 +45,10 @@ public function testCookie()
         $this->getSession()->setCookie('srvr_cookie', null);
         $this->getSession()->reload();
         $this->assertContains('Previous cookie: NO', $this->getSession()->getPage()->getText());
+
+        $this->getSession()->setCookie('semicolon', 'foo;bar;baz');
+        $this->getSession()->visit($this->pathTo('/cookie_page4.php'));
```
Comment (aik099) [inline driver-testsuite/tests/Basic/CookieTest.php]: Does it matter which page is visited at this point? Maybe you can visit any other existing page?

Comment (dawehner) [inline driver-testsuite/tests/Basic/CookieTest.php]: I guess theoretically we could reuse vendor/behat/mink-browserkit-driver/tests/web-fixtures/cookie_page2.php:14 ?

Comment (aik099) [inline driver-testsuite/tests/Basic/CookieTest.php]: You mean https://github.com/minkphp/Mink/blob/master/driver-testsuite/web-fixtures/cookie_page2.php page? There is no line 14 there.

Yes, you can name your cookie `srvr_cookie` and use that page too.

File: driver-testsuite/tests/Basic/CookieTest.php [hunk 2]
```diff
@@ -45,6 +45,11 @@ public function testCookie()
         $this->getSession()->setCookie('srvr_cookie', null);
         $this->getSession()->reload();
         $this->assertContains('Previous cookie: NO', $this->getSession()->getPage()->getText());
+
+        $this->getSession()->setCookie('srvr_cookie', 'foo;bar;baz');
+        $this->getSession()->visit($this->pathTo('/cookie_page2.php'));
+        $this->assertEquals('foo;bar;baz', $this->getSession()->getCookie('srvr_cookie'));
+        $this->assertContains('Previous cookie: foo;bar;baz', $this->getSession()->getPage()->getText());
```
Comment (dawehner) [inline driver-testsuite/tests/Basic/CookieTest.php]: simplified it now

Comment (aik099) [inline driver-testsuite/tests/Basic/CookieTest.php]: Great, 10x.

Comment (stof) [inline driver-testsuite/tests/Basic/CookieTest.php]: shouldn't it be a separate test ?

Comment (aik099) [inline driver-testsuite/tests/Basic/CookieTest.php]: It could be, like `testCookieWithSemicolon`.

Comment (dawehner) [inline driver-testsuite/tests/Basic/CookieTest.php]: Sure, why not

**Comment by dawehner** [comment]
Addressed the feedback from @stof

File: driver-testsuite/tests/Basic/CookieTest.php [hunk 3]
```diff
@@ -47,6 +47,13 @@ public function testCookie()
         $this->assertContains('Previous cookie: NO', $this->getSession()->getPage()->getText());
     }
 
+    public function testCookieWithSemicolon() {
```
Comment (aik099) [inline driver-testsuite/tests/Basic/CookieTest.php]: The opening brace of the function must be on the new line (see other methods in the file).

I recommend you setting coding standard in your IDE and that way this should be reported  or even fixed automatically.

Comment (dawehner) [inline driver-testsuite/tests/Basic/CookieTest.php]: Sure, well, sometimes I'm just too lazy to change it )

**Comment by stof** [comment]
thanks

**Comment by aik099** [comment]
Just for the reference test doesn't pass with Selenium Server 2.48.2 on OSX 10.10.5 with Firefox 41.0.2.

**Pull request merged**